src/strategies/starknet/singleSlotProof.ts

```diff
@@ -168,6 +193,55 @@ export default function createSingleSlotProofStrategy(
           ]
         }
       ];
+    },
+    async getVotingPower(
+      strategyAddress: string,
+      voterAddress: string,
+      timestamp: number,
+      params: string[],
+      clientConfig: ClientConfig
+    ) {
+      const formattedTimestamp = `0x${timestamp.toString(16)}`;
+
+      const key = getStorageVarAddress(timestampToEthBlockNumberStore, formattedTimestamp);
+      let storedBlock = await clientConfig.starkProvider.getStorageAt(strategyAddress, key);
+
+      // proposer might not have used SSP strategy, use latest L1 block
+      if (storedBlock === '0x0') {
+        storedBlock = await clientConfig.starkProvider.getStorageAt(
+          fossilL1HeadersStoreAddress,
+          getStorageVarAddress(latestL1BlockStore)
+        );
+      }
+
+      const block = parseInt(storedBlock as string, 16) - 1;
+
+      try {
+        const strategyContract = new Contract(
+          SingleSlotProofAbi,
+          strategyAddress,
+          clientConfig.starkProvider
+        );
+
+        const proofInputs = await fetchProofInputs(
+          params[0],
+          voterAddress,
+          params[1],
+          block,
+          clientConfig
+        );
+
+        const { voting_power } = await strategyContract.getVotingPower(
+          formattedTimestamp,
+          { value: voterAddress },
+          params,
+          proofInputs.storageProofs[0]
+        );
+
+        return BigInt(voting_power.low) + (BigInt(voting_power.high) << BigInt(128));
+      } catch (err) {
+        return 0n;
```

Comment:
I think we should not catch this error, if it fail we should know about it. For some reason this function fail for the test `should compute voting power for user with delegated tokens` with both Brovider and Infura with this error:
```
ERROR GatewayError: Error message: SingleSlotProof: Invalid slot proof provided
```
For some reason Infura fail also on `should compute voting power for user with delegated tokens` with the same error. This is going to be problematic in the UI cuz we use brovider, it's better if you can debug this now that you have all context.

Reply:
> I think we should not catch this error, if it fail we should know about it. For some reason this function fail for the test should compute voting power for user with delegated tokens with both Brovider and Infura with this error:

I'm not sure which test case you mean, because we had two with same name (one was missing `no` in the name, updated it now). I don't think we can remove this catch, because that's how we know if user has or doesn't have VP. I'm not sure if it's the way it's supposed to work, but strategy contract throws if you call `getVotingPower` for user without voting power - it doesn't return 0.

> Infura fail also on should compute voting power for user with delegated tokens with the same error. This is going to be problematic in the UI cuz we use brovider, it's better if you can debug this now that you have all context.

Infura doesn't fail for me, all tests pass as they should, Brovider doesn't work because of this: https://discord.com/channels/955773041898573854/1031838169282392144/1078299607735476305
bug in Alchemy API.

Reply:
Ok I see, then I think at least the `try` should come just before `await strategyContract.getVotingPower` and not include `fetchProofInputs`